modules/fluid_properties/src/userobjects/MethaneFluidProperties.C

```diff
@@ -229,21 +207,31 @@ MethaneFluidProperties::rho_mu_dpT(Real pressure,
 }
 
 Real
-MethaneFluidProperties::k(Real pressure, Real temperature) const
+MethaneFluidProperties::k(Real /*pressure*/, Real temperature) const
 {
-  Real rho = this->rho(pressure, temperature);
-  return this->k_from_rho_T(rho, temperature);
+  // Check the temperature is in the range of validity (200 K <= T <= 1000 K)
+  if (temperature <= 200.0 || temperature >= 1000.0)
+    throw MooseException("Temperature " + Moose::stringify(temperature) +
+                         "K out of range (200K, 1000K) in " + name() + ": k()");
+
+  const std::vector<Real> a{-1.3401499e-2,
+                            3.663076e-4,
+                            -1.82248608e-6,
+                            5.93987998e-9,
+                            -9.1405505e-12,
+                            6.7896889e-15,
+                            -1.95048736e-18};
+
+  Real kt = 0.0;
+  for (std::size_t i = 0; i < a.size(); ++i)
+    kt += a[i] * std::pow(temperature, i);
```

Comment:
You should use `MathUtils::pow` here instead:
https://github.com/idaholab/moose/blob/devel/framework/src/utils/MathUtils.C#L211

We've measured substantial improvements when we have integer powers.

Reply:
Thanks Cody - I hadn't seen this version before (I must admit it has taken me several minutes to work out what is going on in this one!). I'm always happy to do anything that makes these parts of the calculation faster.

There are potentially a lot of places that I can use this in `fluid_properties`. If it's ok, I'll add a new issue to do this so that I can do all of the fluids at once.

Reply:
Yeah that's fine